topsort-api-v2.yml

```diff
@@ -466,6 +497,236 @@ components:
             The string for a given error may change over time; code should not parse or dispatch based on particular values for this field.
+    Placement:
+      type: object
+      required:
+        - path
+      properties:
+        path:
+          type: string
+          description: >
+            URL path of the page triggering the event.
+            This can be obtained in JS using `window.location.pathname`.
+          example: /categories/dairy
+        position:
+          type: integer
+          description: >
+            For components with multiple items
+            (i.e. search results, similar products, etc),
+            this should indicate the index of a given item within that list.
+          minimum: 0
+        page:
+          type: integer
+          description: >
+            For paginated pages, this should indicate which page number triggered the event.
+          minimum: 0
+        pageSize:
+          type: integer
+          description: >
+            For paginated pages this should indicate how many items are in each result page.
+          minimum: 2
+          example: 15
+    EventsRequest:
+      type: object
+      properties:
+        impressions:
+          type: array
+          items:
+            $ref: '#/components/schemas/Impression'
+          minItems: 1
+        clicks:
+          type: array
+          items:
+            $ref: '#/components/schemas/Click'
+          minItems: 1
+        purchases:
+          type: array
+          items:
+            $ref: '#/components/schemas/Purchase'
+          minItems: 1
+      minProperties: 1
+    Impression:
+      type: object
+      description: >
+        An impression means an ad has become visible to the consumer.
+        In case you cannot send an impression when the product becomes visible, send us an impression event when the product was rendered in the HTML or, if that's also not possible, when your API returns the results.
+        It is important to select the most specific event so that your vendors have more accurate CTR metrics, which will allow them to better predict their campaigns.
+      required:
+        - resolvedBidId
+        - placement
+        - occurredAt
+        - opaqueUserId
+        - id
+      properties:
+        resolvedBidId:
+          type: string
+          description: The `resolvedBidId` field received from the /auctions request.
+          example: WyJiX01mazE1IiwiMTJhNTU4MjgtOGVhZC00Mjk5LTgzMjctY2ViYjAwMmEwZmE4IiwibGlzdGluZ3MiLCJkZWZhdWx0IiwiIl0=
+        placement:
+          $ref: '#/components/schemas/Placement'
+        occurredAt:
+          type: string
+          format: date-time
+          description: RFC3339 formatted timestamp including UTC offset.
+          example: '2009-01-01T12:59:59-05:00'
+        opaqueUserId:
+          type: string
+          description: The opaque user ID allows correlating user activity during a session whether or not they are actually logged in.
```

Comment:
I'd remove the word session and add an explanation saying that this has to be long-lived.